cpp/src/arrow/compute/key_hash_internal.h

```diff
@@ -17,14 +17,11 @@
 
 #pragma once
 
-#if defined(ARROW_HAVE_RUNTIME_AVX2)
-#  include <immintrin.h>
-#endif
-
 #include <cstdint>
 
 #include "arrow/compute/light_array_internal.h"
 #include "arrow/compute/util.h"
+#include "arrow/util/simd.h"
```

Comment:
Is it at all necessary to include this header here?

Reply:
Ping on this.

Reply:
Turns out it is redundant, but solely because it is transitively included in `arrow/compute/util.h`. The code below refers to symbols like `__m256i`. I'm IWYU-ing here. What do you think?